### PR TITLE
Improve OpenAPI schema coverage

### PR DESCRIPTION
**Pull request by vadimkerr**
# What this PR does

Improves OpenAPI schema coverage for internal API:

- Fixes/Improves `alert group` and `feature` endpoints
- Adds `integration` and `user` endpoints

## Which issue(s) this PR fixes

https://github.com/grafana/oncall/issues/3444

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: engine/apps/api/views/alert_group.py [hunk 1]
```diff
@@ -54,101 +58,62 @@ def get_user_queryset(request):
     return User.objects.filter(organization=request.user.organization).distinct()
 
 
-class AlertGroupFilterBackend(filters.DjangoFilterBackend):
```
Comment (vadimkerr) [inline engine/apps/api/views/alert_group.py:57]: Removing this as it doesn't work (the number of SQL queries is the same without this code)

File: engine/apps/api/views/alert_group.py [hunk 2]
```diff
@@ -54,101 +58,62 @@ def get_user_queryset(request):
...
-
-
 class AlertGroupFilter(DateRangeFilterMixin, ModelFieldFilterMixin, filters.FilterSet):
     """
     Examples of possible date formats here https://docs.djangoproject.com/en/1.9/ref/settings/#datetime-input-formats
     """
 
     FILTER_BY_INVOLVED_USERS_ALERT_GROUPS_CUTOFF = 1000
 
-    started_at_gte = filters.DateTimeFilter(field_name="started_at", lookup_expr="gte")
```
Comment (vadimkerr) [inline engine/apps/api/views/alert_group.py:89]: Removing outdated fields:
- `started_at_gte`
- `started_at_lte`
- `resolved_at_gte`
- `resolved_at_lte`
- `id__in`

File: engine/apps/api/views/alert_group.py [hunk 3]
```diff
@@ -54,101 +58,62 @@ def get_user_queryset(request):
...
-        fields = [
-            "id__in",
-            "started_at_gte",
-            "started_at_lte",
-            "resolved_at_lte",
-            "is_root",
-            "resolved_by",
-            "acknowledged_by",
-        ]
-
```
Comment (vadimkerr) [inline engine/apps/api/views/alert_group.py:151]: this is not needed, the filters work the same way without this

File: engine/apps/api/views/alert_group.py [hunk 4]
```diff
@@ -305,15 +268,12 @@ class AlertGroupView(
         "preview_template": [RBACPermission.Permissions.INTEGRATIONS_TEST],
     }
 
-    http_method_names = ["get", "post", "delete"]
```
Comment (vadimkerr) [inline engine/apps/api/views/alert_group.py:308]: this is outdated

File: engine/apps/api/views/alert_group.py [hunk 5]
```diff
@@ -713,12 +698,32 @@ def unpage_user(self, request, pk=None):
...
+            },
+            many=True,
+        )
+    )
     @action(methods=["get"], detail=False)
     def filters(self, request):
         """
         Retrieve a list of valid filter options that can be used to filter alert groups
         """
-        filter_name = request.query_params.get("search", None)
```
Comment (vadimkerr) [inline engine/apps/api/views/alert_group.py:721]: this is outdated/not used by API clients

File: engine/apps/api/views/alert_receive_channel.py [hunk 6]
```diff
@@ -287,10 +356,22 @@ def get_alert_to_template(self, payload=None):
...
+                "href": serializers.CharField(),
+                "global": serializers.BooleanField(required=False),
+            },
+            many=True,
+        )
+    )
     @action(methods=["get"], detail=False)
     def filters(self, request):
         organization = self.request.auth.organization
-        filter_name = request.query_params.get("search", None)
```
Comment (vadimkerr) [inline engine/apps/api/views/alert_receive_channel.py:293]: this is outdated/not used by API clients

File: engine/common/api_helpers/paginators.py
Comment (vadimkerr) [inline engine/common/api_helpers/paginators.py:1]: simplifying paginators + adding schema for them

**Review by mderynck — APPROVED** [review]
LGTM, could probably use another quick pass by someone else given the size :slightly_smiling_face:

**Pull request merged**